python/tests/api/expressions/test_str_to_expr.py

```diff
@@ -117,14 +117,32 @@ def test_ternary_condition():
 
 
 def test_precedence():
-    expected_expr = Expressions.and_(Expressions.or_(Expressions.equal("col_a", 1),
-                                                     Expressions.equal("col_b", 2)),
-                                     Expressions.equal("col_c", 3))
+    expected_expr = Expressions.or_(Expressions.equal("col_a", 1),
+                                    Expressions.and_(Expressions.equal("col_b", 2),
+                                                     Expressions.equal("col_c", 3)))
```

Comment:
Isn't it normal for OR to be lower precedence than AND? I think the original case was correct.

I tested this in Spark and Presto:
```sql
select * from rblue.test where id = 1 or id = 2 and data = 'b'
```
Presto:
```
filterPredicate = (("id" IN (BIGINT '1', BIGINT '2')) AND (("id" = BIGINT '1') OR ("data" = CAST('b' AS varchar))))
```
Spark:
```
filters=(id = 1 OR (id = 2 AND data = 'b'))
```

In both of them, AND has higher precedence than OR.

Reply:
precedence in SQL is generally `()`, `AND`, `OR`.  Doesn't that mean the AND should be evaluated then the OR, which it looks like the newer case is doing?

Reply:
You're right! I got it backward and this is actually fixing the problem.